src/internal/connection-provider-routing.js

```diff
@@ -206,36 +207,40 @@ export default class RoutingConnectionProvider extends PooledConnectionProvider
   }
 
   forget (address, database) {
-    if (database || database === '') {
-      this._routingTables[database].forget(address)
-    } else {
-      Object.values(this._routingTables).forEach(routingTable =>
-        routingTable.forget(address)
-      )
-    }
+    this._routingTableRegistry.apply(database, {
+      applyWhenExists: routingTable => routingTable.forget(address),
+      applyWhenDontExists: () => {
+        throw newError(
+          'Could not forget address for a non existing routing table'
+        )
+      }
+    })
 
     // We're firing and forgetting this operation explicitly and listening for any
     // errors to avoid unhandled promise rejection
     this._connectionPool.purge(address).catch(() => {})
   }
 
   forgetWriter (address, database) {
-    if (database || database === '') {
-      this._routingTables[database].forgetWriter(address)
-    } else {
-      Object.values(this._routingTables).forEach(routingTable =>
-        routingTable.forgetWriter(address)
-      )
-    }
+    this._routingTableRegistry.apply(database, {
+      applyWhenExists: routingTable => routingTable.forgetWriter(address),
+      applyWhenDontExists: () => {
+        throw newError(
```

Comment:
Yes, they assumes it always exists when it came this path. I think we could just ignore if doesn't exist since it's not the case something will stop to work if it tries to forget a writer for a non-existing routing table.